lib/base/logger.cpp

```diff
@@ -289,11 +355,37 @@ Log::~Log()
 	for (const Logger::Ptr& logger : Logger::GetLoggers()) {
 		ObjectLock llock(logger);
 
-		if (!logger->IsActive())
+		if (!logger->IsActive()) {
+			continue;
+		}
+
+		if (entry.Severity < logger->GetMinSeverity()) {
 			continue;
+		}
+
+		auto filter (logger->GetObjectFilter());
+
+		if (filter) {
+			if (!m_Involved) {
+				continue;
+			}
+
+			auto indirect (DependencyGraph::GetAllParents({m_Involved}));
+
+			indirect.emplace(m_Involved);
+
+			// TODO(ak): do something useful here
+			for (auto& object : indirect) {
+			}
+
+			ObjectLock lock (filter);
+
+			for (auto& kv : filter) {
+				auto type (Type::GetByName(kv.first));
+			}
+		}
```

Comment:
It will boil down to what exactly will happen here in the end as it doesn't look very finished so far.

I still see the call to `DependencyGraph::GetAllParents()` here critical. As it's inside `if (filter) {` it isn't terrible, but it isn't exactly great either. I'd probably prefer a variant that wouldn't need that for every message.

Reply:
> I still see the call to `DependencyGraph::GetAllParents()` here critical. As it's inside `if (filter) {` it isn't terrible, but it isn't exactly great either. I'd probably prefer a variant that wouldn't need that for every message.

Despite how I've already optimised that data structure? If yes, this leads us back to the unfinished discussion above: https://github.com/Icinga/icinga2/pull/9844#discussion_r1289812204

Reply:
Looking at the new version of `Log::~Log()` you've just pushed, what you're doing there is far from being (almost) free: walking a graph, creating and deleting multiple sets and vector, obtaining multiple locks. If we want to do this, this needs some performance testing, I wouldn't want to merge this without knowing the impact. I mean this feature is intended for installations where there are so many log messages that you want to filter them. so if there is a high cost per message, this will add up.